app/src/main/java/waiter/Connectable/ClientConnection.java

```diff
@@ -20,11 +21,16 @@ public ClientConnection(Socket socket) throws IOException {
     }
 
     public String read() throws IOException {
-        return this.bufferedReader.readLine();
+        StringBuilder requestContents = new StringBuilder();
+        while(!requestContents.toString().contains("\r\n")) {
+            requestContents.append((char) this.bufferedReader.read());
```

Comment:
What does `requestContents` look like at the end? Are we trying to read the header of the request here?

Reply:
I see. Thanks!